Automattic/wp-calypso
# Checkout: Fix type errors in useSiteDomains

#### Changes proposed in this Pull Request

The `useSiteDomains` hook in checkout returns a list of domain objects from `state.sites.domains.items` in the Redux store. When originally implemented, the hook was written in TypeScript, but the wrong type was used for the data it returns (in fact, the type used was not specified correctly so the types didn't work anyway - it would have had to have been `Site.Domain`).

This PR corrects the type returned by that hook.

#### Testing instructions

See the test instructions for https://github.com/Automattic/wp-calypso/pull/56794
